api/src/test/java/org/openmrs/scheduler/tasks/AutoRetireUsersTaskTest.java

```diff
@@ -0,0 +1,146 @@
+import org.openmrs.User;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.jupiter.BaseContextSensitiveTest;
+import org.openmrs.util.OpenmrsConstants;
+import org.openmrs.util.RoleConstants;
+
+import java.util.Collections;
+import java.util.Date;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class AutoRetireUsersTaskTest extends BaseContextSensitiveTest {
+	private static final long ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+	private static final long TWENTY_THREE_HOURS_IN_MILLISECONDS = 23 * 60 * 60 * 1000;
+	private static final long TWO_DAYS_IN_MILLISECONDS = 2 * ONE_DAY_IN_MILLISECONDS;
+	private static final String ONE_DAY_PROPERTY_VALUE = "1";
+	private static final String TWO_DAYS_PROPERTY_VALUE = "2";
+
+	@Mock
+	private UserService userService;
+	private AdministrationService administrationService;
+	private AutoRetireUsersTask autoRetireUsersTask;
+
+	@BeforeEach
+	public void setup() {
+		administrationService = Context.getAdministrationService();
+		autoRetireUsersTask = new AutoRetireUsersTask();
+	}
+
+	@Test
+	public void shouldRetireUsersWhoseInactivityExceedNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		User inactiveUser = getDefaultUser();
+
+		when(userService.getLastLoginTime(any())).thenReturn(String.valueOf(System.currentTimeMillis() - TWO_DAYS_IN_MILLISECONDS));
+		when(userService.getAllUsers()).thenReturn(Collections.singletonList(inactiveUser));
+
+		autoRetireUsersTask.execute();
+
+		verify(userService, atLeastOnce()).retireUser(inactiveUser, AutoRetireUsersTask.AUTO_RETIRE_REASON);
+	}
+
+	@Test
+	public void shouldNotRetireUsersWhoseInactivityDoNotExceedNumberOfDaysToRetire() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, TWO_DAYS_PROPERTY_VALUE));
+
+		User activeUser = getDefaultUser();
+
+		when(userService.getLastLoginTime(any())).thenReturn(String.valueOf(System.currentTimeMillis() - ONE_DAY_IN_MILLISECONDS));
+		when(userService.getAllUsers()).thenReturn(Collections.singletonList(activeUser));
+
+		autoRetireUsersTask.execute();
+
+		verify(userService, never()).retireUser(activeUser, AutoRetireUsersTask.AUTO_RETIRE_REASON);
+	}
+
+	@Test
+	public void shouldNotRetireAlreadyRetiredUsers() {
+		administrationService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS, ONE_DAY_PROPERTY_VALUE));
+
+		User retiredUser = getDefaultUser();
+		retiredUser.setRetired(true);
```

Comment:
Will this test fail if i remove the above line of setRetired(true)?

Reply:
@dicksonmulli did you see the above comment?

Reply:
Do you have a response to the above comment?

Reply:
https://github.com/openmrs/openmrs-core/pull/4650#issuecomment-2168116350

Reply:
If i do not retire the user in this test, and the test still passes, what then would i be testing for? 😊